sklearn/pipeline.py

```diff
@@ -230,6 +230,12 @@ def __getitem__(self, ind):
             return self.named_steps[ind]
         return est
 
+    def _more_tags(self):
+        # hack to make common cases work:
```

Comment:
Is this a hack???

Reply:
Yes? If an imputer is in a pipeline then it handles missing values no matter what happens afterwards. But our current tags don't distinguish between estimators that allow and pass through missing values (like scalers) and imputers which allow missing values but produce data without missing values.
The correct rule is "everything before an imputer needs to allow NaN" I think?